src/System/IO/Resource/Linear/Internal.hs

```diff
@@ -129,50 +140,34 @@ hClose :: Handle %1 -> RIO ()
 hClose (Handle h) = unsafeRelease h
```

Comment:
two lines above (l. 137, but I couldn't comment there in the review pane), we use `Control.pure`, and a few lines before, we use `Control.return`. Also both `openFile` and `openBinaryFile` look very similar, maybe we could factor them as part of this handle functions overhaul?